dataprofiler/tests/profilers/test_categorical_column_profile.py

```diff
@@ -375,6 +373,32 @@ def test_unalikeability(self):
         profile.update(df_categorical)
         self.assertEqual(profile.unalikeability, 2*(10 + 15 + 6)/90)
 
+    def test_top_k_categories_change(self):
+        options = CategoricalOptions()
+        df_series = pd.Series(
+            ["a", "a", "b", "c", "d", "e", "e", "e", "f", "g"])
+        profile = CategoricalColumn(df_series.name, options)
+        profile.update(df_series)
+        self.assertEqual(
+            len(profile.profile['statistics']['categorical_count']), 7)
+        profile._top_k_categories = 6
+        self.assertEqual(
+            len(profile.profile['statistics']['categorical_count']), 6)
+
+        options.top_k_categories = 6
+        df_series = pd.Series(["a", "a", "b", "c", "d"])
+        profile = CategoricalColumn(df_series.name, options)
+        profile.update(df_series)
+        self.assertEqual(
+            len(profile.profile['statistics']['categorical_count']), 4)
+
+        options.top_k_categories = 7
+        df_series = pd.Series(["a", "c", "b", "c", "d"])
+        profile = CategoricalColumn(df_series.name, options)
+        profile.update(df_series)
+        self.assertEqual(
+            len(profile.profile['statistics']['categorical_count']), 4)
+
```

Comment:
ideally these should have comments explaining what each assert is testing so a user knows the intent:
1. `None`
2. top_k < num cat
3. top_k > num_cat

Is the 4th a duplicate of 3? if so we can remove, no?